lib/service/task_service.dart

```diff
@@ -20,27 +21,60 @@ class TaskService {
     }
   }
 
-  Future<List<Task>> readTasks() async {
-    dynamic data = await client.getData(uri: taskApiBase);
-    List<Task> tasks =
-        List<Task>.from(data.map((taskData) => Task.fromJson(taskData)));
-    print(data);
-    return tasks;
-  }
-
-  Future<void> editTask({required Task task}) async {
+  Future<List<Task>> readTasksByStatus(TaskStatus status) async {
     try {
-      await client.putData(
-        uri: '$taskApiBase/${task.id}',
-        body: task.toJson(),
+      dynamic data = await client.getData(
+        uri: '/tasks',
+        queryParams: {'status': Task.getStatusString(status)},
       );
+
+      if (data is List) {
+        List<Task> tasks =
+            List<Task>.from(data.map((taskData) => Task.fromJson(taskData)));
+        return tasks;
+      } else {
+        print('Unexpected response format: $data');
+        return [];
+      }
     } catch (e) {
-      print('Error editing task: $e');
-      throw Exception('Edit Task Failed');
+      print('Error fetching tasks: $e');
+      return [];
     }
   }
 
-  Future<void> completeTask({required Task task}) async {}
+  Future<void> updateTask({
+    required Task task,
+    TaskStatus? newStatus,
+  }) async {
+    try {
+      if (newStatus != null) {
+        task = task.copyWithStatus(newStatus);
+        await client.putData(
+          uri: '$taskApiBase/${task.id}/${Task.getStatusString(newStatus)}',
+          body: task.toJson(),
+          queryParams: {'status': Task.getStatusString(newStatus)},
+        );
+      } else {
+        await client.putData(
+          uri: '$taskApiBase/${task.id}',
+          body: task.toJson(),
+        );
+      }
+    } catch (e) {
+      print('Error updating task: $e');
+      throw Exception('Update Task Failed');
+    }
+  }
 
-  Future<void> deleteTask({required Task task}) async {}
+  Future<void> deleteTask({required Task task}) async {
+    try {
+      task = task.copyWithStatus(TaskStatus.deleted);
+      await client.deleteData(
+        uri: '$taskApiBase/${task.id}/delete',
```

Comment:
Please remove `/delete` from the URI. The fact that it is a `DELETE` HTTP method should make `/delete` at the end redundant.

Reply:
Changed this, had to change it in my backend too